### DISCUSSION TITLE
MongoDB - How can I change order of One-Many relations?

### DISCUSSION THREAD

**Discussion by indicozy** [prisma/prisma#13487] [Q&A] [answered]
Hi everyone, I'm building LMS in which one module can have many lessons. If I want to change the order of lessons in the module, how can I change the order in the list in MongoDB? Should I create some JSON data that will remember the order of lessons or can I use some native features of Prisma/MongoDB?

### Schema
```
model Module {
  id          String   @id @default(auto()) @map("_id") @db.ObjectId
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  name        String
  description String
  slug        String
  lessons     Lesson[]
  Course      Course?  @relation(fields: [courseId], references: [id])
  courseId    String?  @db.ObjectId

  @@unique([courseId, slug])
}

model Lesson {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  name      String
  slug      String
  Module    Module?  @relation(fields: [moduleId], references: [id])
  moduleId  String?  @db.ObjectId
  blocks    Block[]

  @@unique([moduleId, slug])
}
```

**Comment by nurul3101** [comment, marked as answer]
Hey @indicozy 👋 
Each lesson will be associated with a module so you can keep an `order` field in the Lesson model which would define the order of that lesson in that particular module, then you could `orderBy` the `order` field while fetching Lessons.

I am suggesting something like this as a Lesson model:
```
model Lesson {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  name      String
  slug      String
  order     Int
  Module    Module?  @relation(fields: [moduleId], references: [id])
  moduleId  String?  @db.ObjectId
  blocks    Block[]

  @@unique([moduleId, slug])
}
```

**Reply by indicozy** [reply]
Great solution, Thank You!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
